### PR TITLE
Initial documentation on .onConflict, .ignore, and .merge

### PR DESCRIPTION
**Pull request by nicoburns**
Documentation for UPSERT pull request: https://github.com/knex/knex/pull/3763

### CODE REVIEW

File: sections/builder.js [hunk 1]
```diff
@@ -1655,6 +1655,98 @@ export default [
       }
     ]
   },
+  {
+    type: "method",
+    method: "onConflict",
+    example: "insert(..).onConflict(column) / insert(..).onConflict([column1, column2, ...])",
+    description: "Utlized by PostgreSQL and Sqlite databases. When chained onto an insert query, it specifies the columns to be used in `ON CONFLICT` clause. A call to .onConflict should always be followed by either .ignore or .update (otherwise it does nothing). In MySQL this method is noop, and .ignore and .merge can be used without .onConflict.",
```
Comment (tsondergaard) [inline sections/builder.js]: Typo: Utlized => Utilized

Comment (kibertoad) [reply sections/builder.js:1662]: @nicoburns see typo

File: sections/builder.js [hunk 2]
```diff
@@ -1655,6 +1655,98 @@ export default [
...
+        type: "text",
+        content: "Note: The column(s) specified by this method must have a UNIQUE index on them or the INSERT..ON CONFLICT query will fail to execute. When specifying multiple columns, they must have composite UNIQUE index."
+      },
+    ]
+  },
+  {
+    type: "method",
+    method: "ignore",
+    example: "insert(..).onConflict(..).ignore()",
+    description: "Implemented for the PostgreSQL, MySQL, and Sqlite databases. Modifies an insert query, and causes it silently dropped without error if a conflict occurs. Uses INSERT IGNORE in MySQL, and adds an ON CONFLICT (columns) DO NOTHING clause to the insert statement in PostgreSQL and Sqlite.",
```
Comment (kibertoad) [inline sections/builder.js:1674]: `and causes it to be silently dropped without an error`?

File: sections/builder.js [hunk 3]
```diff
@@ -1655,6 +1655,98 @@ export default [
...
+              { email: "jane@example.com", name: "Jane Doe" },
+              { email: "alex@example.com", name: "Alex Doe" },
+            )
+            .onConflict('email')
+            .update()
+        `
+      },
+      {
+        type: "text",
+        content: "It is also possible to specify data to update seperately from the data to insert. This is useful if you only want to update a subset of the columns. For example, you may want to set a 'created_at' column when inserting, but not wish to update it if the row already exists:"
```
Comment (kibertoad) [inline sections/builder.js:1727]: `but would prefer not to update it`

**Comment by nicoburns** [comment]
Review comments fixed.

File: sections/builder.js [hunk 4]
```diff
@@ -1683,6 +1683,98 @@ export default [
...
+            .ignore()
+        `
+      }
+    ]
+  },
+  {
+    type: "method",
+    method: "merge",
+    example: "insert(..).onConflict(..).merge() / insert(..).onConflict(..).merge(updates)",
+    description: "Implemented for the PostgreSQL, MySQL, and Sqlite databases. Modifies an insert query, to turn it into an 'upsert' operation. Uses ON DUPLICATE KEY UPDATE in MySQL, and adds an ON CONFLICT (columns) DO UPDATE clause to the insert statement in PostgreSQL and Sqlite.",
```
Comment (elhigu) [inline sections/builder.js:1722]: Even that the documentation says how the feature is implemented on each DB, this should also specify explicitly how to write crossdb compatible `.merge`. For many people from above it still will not be clear that mysql ignores the column list parameter and that actually to have cross db compatible merge, all the columns in the table must be listed.

Comment (kibertoad) [reply sections/builder.js:1722]: @nicoburns Could you please address this comment?

Comment (nicoburns) [reply sections/builder.js:1691]: @kibertoad Updated with a note on cross db compatibility.

**Review by kibertoad — APPROVED** [review]

**Review by kibertoad — APPROVED** [review]

**Pull request merged**